app/waf/src/controllers/application.controller.ts

```diff
@@ -155,4 +184,115 @@ export class ApplicationController {
+      throw new HttpErrors.UnprocessableEntity(
+        'No service in Application ' + application.id,
+      );
+    }
+
+    let params: {[key: string]: Object} = {
+      application: application,
+    };
+
+    let tenantAssocs = await this.tenantAssociationRepository.find({
+      where: {
+        tenantId: application.tenantId,
+      },
+    });
+
+    if (tenantAssocs.length === 0) {
+      tenantAssocs = await this.tenantAssociationRepository.find({
+        where: {
+          tenantId: 'default',
+        },
+      });
+
+      if (tenantAssocs.length === 0) {
+        throw new HttpErrors.UnprocessableEntity(
+          'No ADC associated with Application ' + application.id,
+        );
+      }
+    }
+
+    let adcs = await this.adcRepository.find({
+      where: {
+        id: tenantAssocs[0].adcId,
+      },
+    });
+
+    if (adcs.length === 0) {
+      throw new HttpErrors.UnprocessableEntity(
+        'Can not find the ADC associated with Application ' + application.id,
+      );
+    }
+    params.adc = adcs[0];
+
+    let serviceId = application.services[0];
+    let service = await this.serviceRepository.findById(serviceId);
+
+    params.service = service;
+
+    if (service.pool) {
+      let pool = await this.poolRepository.findById(service.pool);
+      params.pool = pool;
+
+      if (pool.members.length > 0) {
+        let members = [];
+        for (let member_id of pool.members) {
+          let m = await this.memberRepository.findById(member_id);
+          members.push(m);
+        }
+        params.members = members;
+      } else {
+        params.members = [];
+      }
+    }
+    let rules = [];
+    let wafs = [];
+    if (application.services.length > 0) {
+      let serviceId = application.services[0];
+      let service = await this.serviceRepository.findById(serviceId);
+      if (service) {
+        params.service = service;
+        let endpointpolicyId = service.endpointpolicy;
+        if (endpointpolicyId != null) {
+          let endpointpolicy = await this.endpointpolicyRepository.findById(endpointpolicyId);
+          if (endpointpolicy) {
+            params.endpointpolicy = endpointpolicy;
+            if (endpointpolicy.rules!=null) {
+              for (let ruleId of endpointpolicy.rules) {
+                let rule = await this.ruleRepository.findById(ruleId);
+		            if (rule) {
+                  let wafId = rule.wafpolicy;
+		              if (wafId != null && wafId != "") { 	
+		     	          let waf = await this.wafpolicyRepository.findById(wafId);
+		     	          if (waf) {
+		 		              wafs.push(waf);
+		     	          }
+	       	        }			
+                  rules.push(rule);
+          	    } 
+              }
+              params.rules=rules;
+              params.wafs=wafs; 
+            }
+          }
+          
+        }
+      }
+    } 	
+    let req = new AS3DeployRequest(params);
+    return await this.as3Service.deploy(AS3_HOST, AS3_PORT, req);
```

Comment:
I have a question that there's `try.. catch ` for as3Service.info, why there's no of it for as3Service.deploy?

When would as3Service throw the exception? Do we really need `try .. catch `?